===================================================================
RCS file: /cvsroot/shadow/libmisc/pwdcheck.c,v
retrieving revision 1.3
retrieving revision 1.4
diff -u -r1.3 -r1.4
--- shadow/libmisc/pwdcheck.c	2004/06/02 23:50:10	1.3
+++ shadow/libmisc/pwdcheck.c	2004/11/02 18:46:30	1.4
@@ -34,7 +34,7 @@
 	retcode = pam_acct_mgmt (pamh, 0);
 	if (retcode == PAM_NEW_AUTHTOK_REQD)
 		retcode = pam_chauthtok (pamh, PAM_CHANGE_EXPIRED_AUTHTOK);
-	else if (retcode)
+	if (retcode)
 		goto bailout;
 
 	if (pam_setcred (pamh, 0))
